model/lifecycle.go

```diff
@@ -1644,3 +1651,25 @@ func addNewTasks(ctx context.Context, batchTimeInfo batchTimeTasksAndVariants, v
 
 	return taskIds, nil
 }
+
+func getTaskIdTables(v *Version, p *Project, newPairs TaskVariantPairs) (TaskIdConfig, error) {
+	if v.Requester == evergreen.RepotrackerVersionRequester {
```

Comment:
I was thinking I wanted it to be just for versions that are guaranteed to contain every task defined in the project. My understanding is that git tag versions can specify specific tasks to create.

I've added a new set of requester types {RepotrackerVersionRequester, TriggerRequester} for this.

Reply:
I originally made this comment:
> Is this meant to include anything for which patch_optional doesn't apply? If that's true then git tag versions should also be in this if statement. (Could probably use !IsPatchRequester(v.Requester))
but then saw this separate thread.

There are also `patch_only` tasks so at that point I don't think any of the requesters are guaranteed to be able to create any tasks.

Reply:
in that case then I think what you had was correct. Triggered versions specify a patch alias, so those aren't guaranteed to have all tasks either

Reply:
But they can still add any dependencies, right? `patch_optional` isn't applicable

Reply:
I thought more about this in the aftermath of the reoccurrence of this bug last Wednesday. This time it wasn't a `patch_optional` task but rather a task whose dependency was excluded from patches. Recent changes (https://github.com/evergreen-ci/evergreen/commit/365ddf0e50b93ebcc622781725e300742fd38b1a) allow a user to add a task to a patch when its dependencies are not patchable. So the task exists without its dependencies and we hit the bug of the task id table containing non-existent tasks.

Since this could happen with any kind of version (generated repotracker tasks can specify dependencies on tasks that are `patch_only`, for instance) I think we need to remove the exception to make sure all the tasks in the table exist/will exist in the version.